### PR TITLE
Adds a new traitor item: The IA-C01G AORTA 'Coral' Generator mech equipment

### PR DESCRIPTION
**Pull request by cowbot92**
All spriting done by @cuackles 

# Document the changes in your pull request

Adds a new 10TC traitor item the The IA-C01G AORTA, a mech equipment piece which can take up a slot on the mech, allowing you to move significantly faster in all mecha types at the cost of greater energy.

Does not work with the gygax or any subtypes

# Why is this good for the game?

This is the final thing I wanted to add to mechs, to really round out the 'mech' identity for a traitor roboticist without giving them their own unique mech.

Bonus: it has a glowy afterimage trail

image

# Testing

Its basically a repurposed gygax leg actuator, although I removed the damage part when running + the shove/punch stuff

After testing I found that im basically going to restrict it from being on gygax (it doesnt stack and can break stuff) but the other mechs work fine

https://github.com/user-attachments/assets/41aa0a0a-1e3f-488e-ba0a-2cfff58a351a

# Spriting

image

# Wiki Documentation

image

TBA

# Changelog

:cl:
rscadd: Added new traitor mech item: The IA-C01G AORTA
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
cool

File: code/game/mecha/coral_gen.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/obj/item/mecha_parts/mecha_equipment/coral_generator
+	name = "IA-C01G AORTA"
+	desc = "A highly classified emergent technology, burns raw redspace crystal to emit a red glow, greatly increasing the mech's movement speed, at the cost of greater energy usage."
```
Comment (ToasterBiome) [inline code/game/mecha/coral_gen.dm]: it specifically burns the dust TO glow red?

File: code/game/mecha/coral_gen.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/obj/item/mecha_parts/mecha_equipment/coral_generator
```
Comment (iamthesilly) [inline code/game/mecha/coral_gen.dm]: This file belongs in the equipment folder somewhere.

File: code/game/mecha/equipment/tools/other_tools.dm [hunk 3]
```diff
@@ -626,3 +626,32 @@
 	if(chassis)
 		chassis.ejection_distance -= ejection_distance
 	. = ..()
+
+////coral generator////
+
+
+
+/datum/action/innate/mecha/mech_overload_mode/Activate(forced_state = null)
```
Comment (iamthesilly) [inline code/game/mecha/equipment/tools/other_tools.dm]: Why is this here?

File: code/game/mecha/mecha.dm [hunk 4]
```diff
@@ -132,6 +132,7 @@
 	var/defence_mode = FALSE
 	var/defence_mode_deflect_chance = 15
 	var/leg_overload_mode = FALSE
+	var/coral_leg_overload_mode = FALSE
```
Comment (iamthesilly) [inline code/game/mecha/mecha.dm:127]: This variable can be moved to the action itself, it isn't used anywhere outside of it.

**Review by SapphicOverload — DISMISSED** [review]

**Comment by Addust** [comment]
holy fuck yogs mechs are getting even more steroids
- first mech suits
- then traitor manuals
- now the coral

outrun a meth addict today in your 2008 tungsten cinderblock

File: code/game/mecha/coral_gen.dm (hunk 1 above)
Comment (cowbot92) [reply code/game/mecha/coral_gen.dm]: it burns the crystal (in lore, no actual crystals are needed) to speed it up & make it schmove fast, I have to hit up molti to see if I can make the after-image only red

File: code/game/mecha/coral_gen.dm (hunk 2 above)
Comment (cowbot92) [reply code/game/mecha/coral_gen.dm]: will move it

File: code/game/mecha/coral_gen.dm (hunk 1 above)
Comment (cowbot92) [reply code/game/mecha/coral_gen.dm]: I guess I should reword it to say it's more like an after-effect

File: code/game/mecha/equipment/tools/other_tools.dm (hunk 3 above)
Comment (cowbot92) [reply code/game/mecha/equipment/tools/other_tools.dm]: That was from an old version, removed!

File: code/game/mecha/coral_gen.dm (hunk 2 above)
Comment (cowbot92) [reply code/game/mecha/coral_gen.dm]: moved

File: code/game/mecha/mecha.dm (hunk 4 above)
Comment (cowbot92) [reply code/game/mecha/mecha.dm:127]: it doesn't like it when I move it to the action unfortunately.

File: code/game/mecha/coral_gen.dm (hunk 1 above)
Comment (cowbot92) [reply code/game/mecha/coral_gen.dm]: Description adjusted

File: code/game/mecha/mecha.dm (hunk 4 above)
Comment (cowbot92) [reply code/game/mecha/mecha.dm:127]: @SapphicOverload

**Review by Chubbygummibear — APPROVED** [review]
because you asked me to

**Review by Chubbygummibear — APPROVED** [review]
cool

File: code/game/mecha/coral_gen.dm (hunk 1 above)
Comment (ToasterBiome) [reply code/game/mecha/coral_gen.dm]: This still doesnt make sense. It doesnt burn dust to emit a red glow right?? That's just sorta an after effect of being evil as fuck right!?!?

Comment (cowbot92) [reply code/game/mecha/coral_gen.dm]: It burns the dust to enhance it's movement, a side effect of burning the dust is the after-image which is evil and red.

**Review by ToasterBiome — APPROVED** [review]
looks good
